astropy/tests/tests/test_skip_remote_data.py

```diff
@@ -6,20 +6,45 @@
 
 from ..helper import remote_data
 from ..helper import pytest
+from ...utils.data import get_pkg_data_filename, download_file
 
 
 @remote_data
 def test_skip_remote_data(pytestconfig):
-    # this test was called from the command line and it should behave as if
-    # astropy.test() has remote_data=True
-    if not hasattr(pytestconfig.option, 'remotedata'):
-        assert True
 
-    # astropy.test() has remote_data=False but we still got here somehow,
+    # astropy.test() has remote_data=none or remote_data=astropy but we still
+    # got here somehow, so fail with a helpful message
+
+    if pytestconfig.getoption('remote_data') == 'none':
+        pytest.fail('@remote_data was not skipped with remote_data=none')
+    elif pytestconfig.getoption('remote_data') == 'astropy':
+        pytest.fail('@remote_data was not skipped with remote_data=astropy')
+
+    # Test Astropy URL
+    get_pkg_data_filename('galactic_center/gc_2mass_k.fits')
+
+    # Test non-Astropy URL
+    download_file('http://www.google.com')
+
+
+@remote_data(source='astropy')
+def test_skip_remote_data_astropy(pytestconfig):
+
+    # astropy.test() has remote_data=none but we still got here somehow,
     # so fail with a helpful message
-    elif not getattr(pytestconfig.option, 'remotedata'):
-        pytest.fail('@remote_data was not skipped with remote_data=False')
 
-    # astropy.test() has remote_data=True, so pass
-    elif getattr(pytestconfig.option, 'remotedata'):
-        assert True
+    if pytestconfig.getoption('remote_data') == 'none':
+        pytest.fail('@remote_data was not skipped with remote_data=none')
+    else:
```

Comment:
This `else` clause is superfluous, no? The same package data gets downloaded on line 42.